include/albatross/src/cereal/eigen.hpp

```diff
@@ -147,6 +147,74 @@ inline void load_lower_triangle(Archive &archive,
   }
 }
 
+template <class Archive, class _Scalar, int _Options, typename _StorageIndex>
+inline void save(Archive &ar,
+                 Eigen::SparseMatrix<_Scalar, _Options, _StorageIndex> const &m,
+                 const std::uint32_t version ALBATROSS_UNUSED) {
+  const _StorageIndex rows = m.rows();
+  const _StorageIndex cols = m.cols();
+  const std::size_t nnz = albatross::cast::to_size(m.nonZeros());
+  const std::size_t size_bytes = nnz * sizeof(_Scalar);
+
+  std::vector<Eigen::Triplet<_Scalar>> elems;
+  elems.reserve(nnz);
+  for (_StorageIndex col = 0; col < m.outerSize(); ++col) {
+    for (typename Eigen::SparseMatrix<_Scalar, _Options,
+                                      _StorageIndex>::InnerIterator it(m, col);
+         it; ++it) {
+      elems.emplace_back(it.col(), it.row(), it.value());
+    }
+  }
+
+  std::string payload =
+      gzip::compress(reinterpret_cast<const char *>(elems.data()), size_bytes);
```

Comment:
I guess we can followup, but curious if this should end up using zstd?

Reply:
See JM's comment above!